components/core/tests/test-kql.cpp

```diff
@@ -0,0 +1,193 @@
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <Catch2/single_include/catch2/catch.hpp>
+#include <spdlog/spdlog.h>
+
+#include "../src/clp_s/search/AndExpr.hpp"
+#include "../src/clp_s/search/FilterExpr.hpp"
+#include "../src/clp_s/search/kql/kql.hpp"
+#include "../src/clp_s/search/OrExpr.hpp"
+
+using clp_s::search::AndExpr;
+using clp_s::search::DescriptorToken;
+using clp_s::search::FilterExpr;
+using clp_s::search::FilterOperation;
+using clp_s::search::kql::parse_kql_expression;
+using clp_s::search::OrExpr;
+using std::string;
+using std::stringstream;
+using std::vector;
+
+TEST_CASE("Test parsing KQL", "[KQL]") {
+    // Suppress logging
+    auto previous_logging_level = spdlog::default_logger()->level();
+    spdlog::default_logger()->set_level(spdlog::level::off);
```

Comment:
If an assertion fails below, this will actually end up leaving logging off for all remaining test cases. Can we add a `LogSuppressor` class like this:

```cpp
/**
 * A class that suppresses logs so long as it instantiated.
 */
class LogSuppressor {
public:
    LogSuppressor() {
        m_previous_logging_level = spdlog::default_logger()->level();
        spdlog::default_logger()->set_level(spdlog::level::off);
    }
    LogSuppressor(LogSuppressor const& other) = default;
    LogSuppressor(LogSuppressor&& other) noexcept = default;
    
    LogSuppressor& operator=(LogSuppressor const& other) = default;
    LogSuppressor& operator=(LogSuppressor&& other) noexcept = default;

    ~LogSuppressor() { spdlog::default_logger()->set_level(m_previous_logging_level); }

private:
    spdlog::level::level_enum m_previous_logging_level;
};
```

And then replace the current log suppression code with: `LogSuppressor{};`

Reply:
Sure. Put it in its own file under tests for re-use?

Reply:
Sgtm